pkg/controllers/c2cc/probe.go

```
@@ -0,0 +1,228 @@
+
+	msClient, err := microshiftclientset.NewForConfig(restCfg)
+	if err != nil {
+		return fmt.Errorf("failed to create microshift client: %w", err)
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, "ok")
+	})
+	server := &http.Server{Addr: ":8080", Handler: mux}
```

golangci-lint (2.12.2)

[medium] G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server

(gosec)